routing/payment_lifecycle_test.go

```diff
@@ -526,13 +553,18 @@ func TestRouterPaymentStateMachine(t *testing.T) {
 				routerFailAttempt,
 				routerFailAttempt,
 
+				// shard 3
```

Comment:
Put this back to dispatching all shards first, but did make some other changes - explained [here](https://github.com/lightningnetwork/lnd/pull/5214#issuecomment-821067208). I think the change is ok because it still covers the "settles less than total amount" case?